source/kernel/src/services/emb_display.rs

```diff
@@ -222,24 +351,91 @@ impl DrawTarget for FrameChunk {
             let Ok((x, y)): Result<(u32, u32) , _> =  coord.try_into() else {
                 continue;
             };
-            if x >= self.width {
-                continue;
-            }
-            if y >= self.height {
-                continue;
-            }
-
-            let index: u32 = x + y * self.width;
-            // TODO: Implement bound checks and return BufferFull if needed
-            self.bytes.as_slice_mut()[index as usize] = color.luma();
+            self.draw_pixel(x, y, color.is_on());
         }
 
         Ok(())
     }
 }
 
+/// This is used for placing the [FrameChunk] in the overall display
+pub struct FrameLocSize {
+    /// Offset in pixels from the top left corner (rightward)
+    pub offset_x: u32,
+    /// Offset in pixels from the top left corner (downward)
+    pub offset_y: u32,
+    /// Width of the frame chunk in pixels
+    pub width: u32,
+    /// Height of the frame chunk in pixels
+    pub height: u32,
+}
+
+/// Kinds of [FrameChunk]s
+#[derive(Clone, Copy, Debug)]
+#[non_exhaustive]
+pub enum FrameKind {
+    /// Monochrome - matches [FrameChunk::Mono]
+    Mono,
+}
+
+impl OriginDimensions for MonoChunk {
+    fn size(&self) -> Size {
+        Size::new(self.meta.width, self.meta.height)
+    }
+}
+
 impl OriginDimensions for FrameChunk {
+    #[inline]
     fn size(&self) -> Size {
-        Size::new(self.width, self.height)
+        match self {
+            FrameChunk::Mono(mc) => mc.size(),
+        }
+    }
+}
+
+/// Metadata of the entire display
+#[derive(Copy, Clone, Debug)]
+pub struct DisplayMetadata {
+    pub kind: FrameKind,
+    pub width: u32,
+    pub height: u32,
+}
+
+#[derive(Copy, Clone)]
+pub struct FrameChunkMetadata {
+    start_x: u32,
+    start_y: u32,
+    width: u32,
+    height: u32,
+}
+
+/// Metadata of the [FrameChunk]
```

Comment:
tiny nit: i think you meant:
```suggestion
/// Metadata of the [FrameChunk]
#[derive(Copy, Clone)]
pub struct FrameChunkMetadata {
    start_x: u32,
    start_y: u32,
    width: u32,
    height: u32,
}

```